pkg/testutils/testserver.go

```diff
@@ -0,0 +1,180 @@
+package testutils
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	client "k8s.io/client-go/kubernetes"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"k8s.io/kubernetes/pkg/controlplane"
+	"k8s.io/kubernetes/test/integration/framework"
+
+	"sigs.k8s.io/aws-iam-authenticator/pkg/config"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/config/certs"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/config/kubeconfig"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/mapper"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/server"
+)
+
+const (
+	hardcodedHealthcheckPort         = 21363
+	hardcodedAuthenticatorServerPort = 21362
+)
+
+// AuthenticatorTestFrameworkSetup holds configuration information for a kube-apiserver test server.
+type AuthenticatorTestFrameworkSetup struct {
+	ModifyAuthenticatorServerConfig func(*config.Config)
+	AuthenticatorClientBinaryPath   string
+}
+
+func StartAuthenticatorTestFramework(t *testing.T, stopCh <-chan struct{}, setup AuthenticatorTestFrameworkSetup) (client.Interface, client.Interface) {
+	cfg, err := testConfig(t, setup)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := cfg.GetOrCreateCertificate(); err != nil {
```

Comment:
where do these files get written? to pwd ?

Reply:
They get written to the "stateDir".